library/java/net/openid/appauth/TokenResponse.java

```diff
@@ -216,7 +216,10 @@ public Builder fromResponseJson(@NonNull JSONObject json) throws JSONException {
             setTokenType(JsonUtil.getString(json, KEY_TOKEN_TYPE));
             setAccessToken(JsonUtil.getStringIfDefined(json, KEY_ACCESS_TOKEN));
             if (json.has(KEY_EXPIRES_AT)) {
-                setAccessTokenExpirationTime(json.getLong(KEY_EXPIRES_AT));
+                long expiresAt = json.optLong(KEY_EXPIRES_AT);
+                if (expiresAt != 0) {
+                    setAccessTokenExpirationTime(expiresAt);
+                }
```

Comment:
getLongIfDefined with a null check would more accurately handle this, though I agree that in practice an `expires_at` value of zero isn't likely.